orttraining/orttraining/python/training/ortmodule/_io.py

```diff
@@ -40,11 +56,20 @@ def __repr__(self) -> str:
             \t#Positionals (non-None): {self.num_positionals_non_none}
             \tKeyword names:           {self.keyword_names}'''
 
-    def flatten(self, args, kwargs):
+    def flatten(self, args, kwargs, device):
         '''Flatten args and kwargs in a single tuple of tensors with strict ordering'''
 
-        ret = list(args)
+        ret = [None]*len(args)
+        for idx, arg in enumerate(args):
+            if _Primitive.is_primitive(arg):
+                ret[idx] = _Primitive.get_tensor(arg, device)
+            else:
+                ret[idx] = arg
         ret += [kwargs[name] for name in self.names if name in kwargs]
+        for idx in range(len(args), len(ret)):
+            if _Primitive.is_primitive(ret[idx]):
+                ret[idx] = _Primitive.get_tensor(ret[idx], device)
```

Comment:
I think my previous comment is just answered here, but I want to double check. This is wrapping kwargs with primitive types, right?

Use list comprehension to get better perf. List comprehension can "predict"the number of iterations and plan allocation ahead of time better than plain loop

Reply:
I will try to use a list comprehension instead.